tests/contest/contest/src/tests/kill_no_effect/kill_no_effect_test.rs

```diff
@@ -0,0 +1,105 @@
+use std::time::Duration;
+
+use anyhow::{Context, Result, anyhow};
+use oci_spec::runtime::{ProcessBuilder, Spec, SpecBuilder};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::tests::lifecycle::ContainerLifecycle;
+use crate::utils::get_state;
+
+fn create_spec(args: &[&str]) -> Result<Spec> {
+    let args_vec: Vec<String> = args.iter().map(|&a| a.into()).collect();
+    let spec = SpecBuilder::default()
+        .process(
+            ProcessBuilder::default()
+                .args(args_vec)
+                .build()
+                .context("failed to build process spec")?,
+        )
+        .build()
+        .context("failed to build spec")?;
+    Ok(spec)
+}
+
+fn failed_and_delete(text: String, container: ContainerLifecycle) -> TestResult {
+    let delete_result = container.delete();
+    match delete_result {
+        TestResult::Passed => TestResult::Failed(anyhow!(text)),
+        TestResult::Failed(err) => TestResult::Failed(anyhow!(
+            "{}; also container deletion failed: {:?}",
+            text,
+            err
+        )),
+        _ => TestResult::Failed(anyhow!("{}; unexpected delete result", text)),
+    }
+}
+
+//This test MUST ensure that attempting to send a signal to a container that is neither created nor running has no effect on the container and generates an error.
+fn kill_no_effect_test() -> TestResult {
+    let container = ContainerLifecycle::new();
+    let spec = create_spec(&["sleep", "1"]).unwrap();
+
+    match container.create_with_spec(spec) {
+        TestResult::Passed => {}
+        _ => return failed_and_delete("Failed to recreate container".to_string(), container),
+    }
+
+    match container.start() {
+        TestResult::Passed => {}
+        _ => return failed_and_delete(("Failed to start container").to_string(), container),
+    }
+
+    container.wait_for_state("stopped", Duration::from_secs(5));
+
+    // get state before kill
+    let (before_stdout, before_stderr) =
```

Comment:
[MEMO]
Please leave this review comment unresolved for now.
This is something I also had in mind, but we don’t need to address it here.

If `get_state` returned `Err` on failure, I think we could write it like this:

https://github.com/youki-dev/youki/blob/main/tests/contest/contest/src/utils/test_utils.rs#L116

```rust
let before_stdout = match get_state(container.get_id(), container.get_project_path()) {
    Ok(v) => v,
    Err(e) => {
        return failed_and_delete(
            format!("Failed to get container state before kill: {e}"),
            container,
        )
    }
};
```

This feels more idiomatic in Rust to me (does the current approach feel more Go-like?).